app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,15 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      req_class_name = @req.to_hash[:fullpath]
+                           .split('?').last.split('&')                        # '?' for query, '&' connects parameters
+                           .select { |param| param.starts_with?("ems_type") } # 'parameters with "ems_type" name
+                           .first.split('=').last                             # choosing only the first occuracne.
+      ems_class = ExtManagementSystem.descendants.select { |ems| ems.ems_type == req_class_name }.first
+      render :json => ems_class.respond_to?(:provider_settings) ? ems_class.provider_settings : {}
```

Comment:
I think you will want to use `render_options(:providers, <your provider settings hash)` here. This will return all of the default options that are passed back from `OPTIONS /providers`, and append your provider settings hash to it.